### ISSUE TITLE
Check if Data Integrity is working in the Finetuning BERT example?

### ISSUE THREAD

**Issue by vipgupta** [uptrain-ai/uptrain#123]
In the [finetuning BERT example](https://github.com/uptrain-ai/uptrain/blob/main/examples/finetuning_LLM/run.ipynb), we want to verify if the UpTrain framework is catching null values. 

According to this [check](https://github.com/uptrain-ai/uptrain/blob/main/uptrain/core/classes/anomalies/data_integrity.py#L24), it should work. 

@a-r-r-o-w assigning this to you since you were the author. Happy to discuss and help!

**Comment by a-r-r-o-w** [comment]
I seem to have located the cause of the issue by printing values [here](https://github.com/uptrain-ai/uptrain/blob/e06c79ed28f937f61b20e72389382b22f7f6de8c/uptrain/core/classes/measurables/measurable.py#L16) .

```
inputs: {'text': ['nan'], 'id': [12], 'output': [None], 'gt': [None]}

outputs: [None]

gts: [None]

extra: {'id': [12]}

vals: ['nan']
```

When the text in the generated dataset is `None` (see [here](https://github.com/uptrain-ai/uptrain/blob/e06c79ed28f937f61b20e72389382b22f7f6de8c/examples/finetuning_LLM/helper_funcs.py#L112)), vals is the list `['nan']` and so the data integrity check `has_issue = signal_value == None` fails. I'll look into how this could be fixed.

**Comment by a-r-r-o-w** [comment]
The cause of the issue is that the `uptrain/core/classes/framework.py` implicitly _requires_ `inputs` to be of type `<class numpy.ndarray>` so that the [checks](https://github.com/uptrain-ai/uptrain/blob/e06c79ed28f937f61b20e72389382b22f7f6de8c/uptrain/core/classes/anomalies/data_integrity.py#L20) in `uptrain/core/classes/anomalies/data_integrity.py` work as expected.

Here, when `inputs['text']` is any value (example `['Hi, this is a sentence']`, or even `['nan']`), `signal_value` ends up having type `<class list>` and therefore the comparison `signal_value == None` will only check if the list is None/empty instead of the numpy overload meaning.

```py
# 1
signal_value = ['Hi, this is a sentence', None, 'Another sentence']
print(signal_value == None) # False

# 2
signal_value = [None]
print(signal_value == None) # False

# 3
signal_value = np.array(['Hi, this is a sentence', None, 'Another sentence'])
print(signal_value == None) # [False, True, False]

# 4 
signal_value = [None]
print(signal_value == None) # [True]
```

(3) and (4) is what is expected to happen in the logic but (1) and (2) is what happens when `inputs` is not a numpy array. I've added the fix for the same in #126.

**Comment by vipgupta** [comment]
Solved by #126, courtesy @a-r-r-o-w.
